Build failure new in v26 with default-features = false

**Description**
Build of `wgpu` fails with `default-features = false, features = ["webgl"]`

**Repro steps**
```bash
git checkout v26.0.0 # or trunk
cargo check -p wgpu --no-default-features  --features webgl --target wasm32-unknown-unknown
```

**Expected vs observed behavior**
```
warning: unused import: `num_traits::float::FloatCore`
  --> naga/src/back/pipeline_constants.rs:22:5
   |
22 | use num_traits::float::FloatCore as _;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default

warning: `naga` (lib) generated 1 warning
    Checking wgpu v26.0.0 (...)
error[E0425]: cannot find value `desired_adapter_name` in this scope
  --> wgpu/src/util/init.rs:35:47
   |
35 |         if info.name.to_lowercase().contains(&desired_adapter_name) {
   |                                               ^^^^^^^^^^^^^^^^^^^^ not found in this scope

warning: unreachable statement
  --> wgpu/src/util/init.rs:23:5
   |
19 |             return Err(wgt::RequestAdapterError::EnvNotSet);
   |             ----------------------------------------------- any code following this expression is unreachable
...
23 |     let adapters = instance.enumerate_adapters(crate::Backends::all());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unreachable statement
   |
   = note: `#[warn(unreachable_code)]` on by default

For more information about this error, try `rustc --explain E0425`.
warning: `wgpu` (lib) generated 1 warning
error: could not compile `wgpu` (lib) due to 1 previous error; 1 warning emitted
```

**Platform**
rustc 1.84.1
wgpu v26.0.0
target=wasm32-unknown-unknown

--- a/wgpu/src/util/init.rs
+++ b/wgpu/src/util/init.rs
@@ -5,20 +5,23 @@ use crate::Backends;
 
 /// Initialize the adapter obeying the `WGPU_ADAPTER_NAME` environment variable.
 #[cfg(wgpu_core)]
+#[cfg_attr(not(std), expect(unused_variables, unreachable_code))]
 pub fn initialize_adapter_from_env(
     instance: &Instance,
     compatible_surface: Option<&Surface<'_>>,
 ) -> Result<Adapter, wgt::RequestAdapterError> {
-    cfg_if::cfg_if! {
-        if #[cfg(std)] {
-            let desired_adapter_name = std::env::var("WGPU_ADAPTER_NAME")
-                .as_deref()
-                .map(str::to_lowercase)
-                .map_err(|_| wgt::RequestAdapterError::EnvNotSet)?;
-        } else {
-            return Err(wgt::RequestAdapterError::EnvNotSet);
+    let desired_adapter_name: alloc::string::String = {
+        cfg_if::cfg_if! {
+            if #[cfg(std)] {
+                std::env::var("WGPU_ADAPTER_NAME")
+                    .as_deref()
+                    .map(str::to_lowercase)
+                    .map_err(|_| wgt::RequestAdapterError::EnvNotSet)?
+            } else {
+                return Err(wgt::RequestAdapterError::EnvNotSet)
+            }
         }
-    }
+    };
 
     let adapters = instance.enumerate_adapters(crate::Backends::all());
 

